src/main/scala/FarmRelated/Farmer.scala

```diff
@@ -484,13 +485,32 @@ class Farmer(_s: Simulation, _obs: Observator, _landAdmin: LandAdministrator, _a
        * Same logic as SwissLand
        * @return true if Farmer has a child, and household incomes are slightly above a regional average
        */
-      def childShouldInherit: Boolean = children.nonEmpty && last5HouseHoldIncomes.head > 1.01 * regionalAverageHouseHoldIncomes
+      def childrenShouldInherit: Boolean = children.nonEmpty && children.filter(c => c.want_take_over && c.gender == "male").nonEmpty //&& last5HouseHoldIncomes.head > 1.01 * regionalAverageHouseHoldIncomes
 
       /** Keeps all the herds, crops, contact network,... So just reset the age to 35 (assume) and give a child with probability 0.875 (respect probability used by SwissLand)*/
-      def transferToChild: Unit = {
+      def transferToChildren: Unit = {
         //no problem like in Terminator where John Connor is older than Sarah Connor cause if exit before 65, this is because
         //house holds are negative, thus son will not take over (only if all regional incomes are negative but in this case there is a problem)
-        age = CONSTANTS.CHILD_TAKE_OVER_AGE
+        val nbParcelsPerChild = parcels.length / children.length
+        var remaining_parcels = parcels.length - nbParcelsPerChild
+        val successors = children.filter(_.want_take_over)
+        //println("NB_PARCELS= "+parcels)
+        //println("NB_SUCCESSORS = " + successors.length)
+        successors.foreach((succ: Child) => {
+          val newFarmer = new Farmer(s, obs, landAdmin, succ.age, List())
+          if (remaining_parcels != 0){
+            newFarmer.parcels = parcels.take(nbParcelsPerChild + 1)
+            parcels = parcels.drop(nbParcelsPerChild + 1)
```

Comment:
Need to update remaining_parcels by -1, otw condition  "remaining_parcels != 0" could always be true